sw/host/opentitanlib/src/image/manifest_ext.rs

```diff
@@ -86,11 +93,39 @@ impl ManifestExtSpec {
     pub fn source_path(&self) -> Option<&Path> {
         self.relative_path.as_deref()
     }
+
+    pub fn check_for_duplicates(&self, id: ManifestExtId) -> Result<()> {
+        ensure!(
+            !self
+                .signed_region
+                .iter()
+                .chain(self.unsigned_region.iter())
+                .any(|e| e.id() == u32::from(ManifestExtId::spx_key)),
```

Comment:
nit: should this function check for `id` instead?
```suggestion
                .any(|e| e.id() == id),
```